framework/bundles/org.jboss.tools.rsp.server/src/main/java/org/jboss/tools/rsp/server/model/ServerManagementModel.java

```diff
@@ -171,9 +185,67 @@ protected IJobManager createJobManager() {
 
 	@Override
 	public void dispose() {
+		shutdownAllServers();
 		if( this.jobManager != null ) {
 			this.jobManager.shutdown();
 		}
 	}
 
+	protected void shutdownAllServers() {
+		if( getNotStoppedServers().isEmpty())
+			return;
+		
+		System.out.println("Shutting down started servers");
+		shutdownServers(getStartedServers(), false);
+		System.out.println("Shutting down all servers");
+		shutdownServers(getNotStoppedServers(), true);
+	}
+	
+	protected void shutdownServers(List<IServer> list, boolean force) {
+		if( list.isEmpty()) 
+			return;
+		
+		ExecutorService threadExecutor = Executors.newFixedThreadPool(list.size());
+		CountDownLatch latch = new CountDownLatch(list.size());
+		Iterator<IServer> it = list.iterator();
+		while(it.hasNext()) {
+			final IServer next = it.next();
+			System.out.println("Shutting down " + next.getId());
+
+			threadExecutor.submit( new Runnable() {
+				@Override
+				public void run() {
+					System.out.println("Shutting down 1 " + next.getId());
+					IServerModelListener l = new ServerModelListenerAdapter() {
+						public void serverStateChanged(ServerHandle server, ServerState state) {
+							if( server.getId().equals(next.getId()) && state.getState() == ServerManagementAPIConstants.STATE_STOPPED) {
+								System.out.println("Shutting down 2 " + next.getId());
+								serverModel.removeServerModelListener(this);
+								latch.countDown();
+							}
+						}
+					};
+					serverModel.addServerModelListener(l);
+					next.getDelegate().stop(force);
+				}
+			});
+			try {
+				latch.await(600000, TimeUnit.MILLISECONDS);
```

Comment:
beside the timeout that ideally would better advertised in a constant, I'm wondering if we shouldn't consume the boolean returned by ```await``` and exit with an error?